### ISSUE TITLE
when docker search image for a private registry which require Authentication，it will return 401

### ISSUE THREAD

**Issue by cc272309126** [docker/docker#9922]
Hi，
when docker search image for a private registry which require authentication，it will return 401
reproduce steps
1,create a private docker-registry "docker-registry.xxx" which require authentication.
2,docker search  docker-registry.xxx/centos.
3.it will report http status 401.

I think docker search need support private registry which require auth. if needed I could provide the code and test case.

**Comment by alexdglover** [comment]
Hey cc272309126 - 

Are you using NGINX and htaccess for your authentication? I have the same issue. I don't know if it's intentional or a bug, but there is a workaround:

docker search username:password@docker-registry.xxx/ubuntu

I'm guessing this has something to do with basic authentication, but I don't know why push/pull would work and search does not.

**Comment by cc272309126** [comment]
Hi alexdglover
yes, I ues nginx and htaccess, I read the code. I found pull and push will load the .dockercfg to get the authentication info for private docker-registry. but search will not.

**Comment by dkjer** [comment]
Fix for this: #10050 (tests haven't been added yet)

**Comment by jfrazelle** [comment]
I am going to close this as a duplicate of https://github.com/docker/docker/issues/11016 which has a very good description hope that is ok :)

**Comment by phemmer** [comment]
I don't think this (or any of the issues that were closed as a duplicate of this) are a dup of #11016.
This, and all the issues closed as a duplicate of it are about auth not workking. #11016 is about displaying a proper error, not that private registry auth doesn't work.

In fact the author of #11016 went and created another issue, #11161, which seems like a real dup of this one.

**Comment by pilt** [comment]
If you're letting nginx handle basic auth, you can add something like this to your `nginx.conf` to make `docker search REGISTRY_ADDR/foo` work (allows anonymous users to search):

```
        location /v1/search {
            auth_basic off;
            proxy_pass http://REGISTRY_ADDR:5000;
        }
```
